src/components/forms/ContactForm/ContactForm.tsx

```diff
@@ -0,0 +1,165 @@
+import ReCAPTCHA from "react-google-recaptcha";
+
+import { sendContactForm } from "@/actions";
+import { useShowNotification } from "@/hooks";
+
+import styles from "./ContactForm.module.scss";
+
+const EMAIL_SUBJECT_KEYS = ["aboutSite", "order", "requestTour", "cooperation", "other"] as const;
+
+type EmailSubject = (typeof EMAIL_SUBJECT_KEYS)[number];
+
+interface FormValues {
+  name: string;
+  email: string;
+  subject: string;
+  message: string;
+}
+
+interface Props {
+  reCaptchaSiteKey: string;
+}
+
+const ContactForm: FC<Props> = ({ reCaptchaSiteKey }) => {
+  const [captchaToken, setCaptchaToken] = useState<string | null>(null);
+  const [isCaptchaLoaded, setIsCaptchaLoaded] = useState(false);
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const reCaptchaRef: any = useRef();
+  const t = useTranslations();
+  const showNotification = useShowNotification();
+
+  const form = useForm<FormValues>({
+    initialValues: {
+      name: "",
+      email: "",
+      subject: "",
+      message: "",
+    },
+    validate: {
+      name: isNotEmpty(t("contactForm.fields.name.validation")),
+      email: isNotEmpty(t("contactForm.fields.email.validation")),
+      subject: isNotEmpty(t("contactForm.fields.subject.validation")),
+      message: isNotEmpty(t("contactForm.fields.message.validation")),
+    },
+  });
+
+  const isSubmitEnabled = form.isDirty() && Boolean(captchaToken) && !isSubmitting;
+
+  const handleCaptchaChange = (token: string | null) => {
+    setCaptchaToken(token);
+  };
+
+  const handleSubmit = async (values: FormValues) => {
+    setIsSubmitting(true);
+
+    if (!captchaToken) {
+      showNotification({
+        type: "error",
+        message: t("reCaptcha.message"),
+      });
+
+      reCaptchaRef.current?.focus();
+
+      return;
+    }
+
+    const result = await sendContactForm({
+      ...values,
+      token: captchaToken,
+    });
+
+    if (result.success) {
+      showNotification({
+        type: "success",
+        message: t("action.sendEmail.success"),
+      });
+      form.reset();
+      setCaptchaToken(null);
+      reCaptchaRef.current?.reset();
+    } else {
+      showNotification({
+        type: "error",
+        message: result.message || t("action.sendEmail.error"),
+      });
+    }
+
+    setIsSubmitting(false);
+  };
+
+  const translatedSubjects = useMemo(() => {
+    return EMAIL_SUBJECT_KEYS.map((key: EmailSubject) => t(`contactForm.subjects.${key}`));
+  }, [t]);
+
+  return (
+    <Box mx="auto" className={styles.formContainer}>
+      <Fieldset legend={<h3>{t("contactForm.title")}</h3>} variant="filled">
+        <form className={styles.form} onSubmit={form.onSubmit(handleSubmit)}>
+          <TextInput
+            withAsterisk
+            label={t("contactForm.fields.name.label")}
+            placeholder={t("contactForm.fields.name.placeholder")}
+            key={form.key("name")}
+            {...form.getInputProps("name")}
```

Comment:
Just to confirm: is it safe to put spread operator for `form.getInputProps("name")`? Won't it accidentally override some props you declared earlier? Perhaps for safety it is better to put it in the beginning

Reply:
All the props, specified above `...form.get...` will be safely passed to components. I don't think it should be in the beginning, since devs are who can accidentally erase the necessary props for the form :)